pkg/network/node_identity.go

```diff
@@ -0,0 +1,209 @@
+				Kind:       "Deployment",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+		}
+		nsn := types.NamespacedName{Namespace: hcpCfg.Namespace, Name: "network-node-identity"}
+		if err := client.Default().CRClient().Get(context.TODO(), nsn, webhookDeployment); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to retrieve existing network-node-identity deployment: %w", err)
+			} else {
+				klog.Infof("network-node-identity deployment does not exist")
+			}
+		} else {
+			webhookReady = !deploymentProgressing(webhookDeployment)
+		}
+
+		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, "network/node-identity/managed"))
+	} else {
+		// self-hosted specific
+		data.Data["NetworkNodeIdentityImage"] = os.Getenv("OVN_IMAGE")
+
+		// NetworkNodeIdentityTerminationDurationSeconds holds the allowed termination duration
+		// During node reboot, the webhook has to wait for the API server to terminate first to avoid disruptions
+		data.Data["NetworkNodeIdentityTerminationDurationSeconds"] = 200
+
+		apiServer := bootstrapResult.Infra.APIServers[bootstrap.APIServerDefault]
+		data.Data["K8S_APISERVER"] = "https://" + net.JoinHostPort(apiServer.Host, apiServer.Port)
+
+		// NetworkNodeIdentityAddress is only used in self-hosted deployments where the webhook listens on loopback
+		// listening on localhost always picks the v4 address while dialing to localhost can choose either one
+		// https://github.com/golang/go/issues/9334
+		// For that reason set the webhook address use the loopback address of the primary IP family
+		// Note: ServiceNetwork cannot be empty, so it is safe to use the first element
+		data.Data["NetworkNodeIdentityAddress"] = "127.0.0.1"
+		if utilnet.IsIPv6CIDRString(conf.ServiceNetwork[0]) {
+			data.Data["NetworkNodeIdentityAddress"] = "[::1]"
+		}
+
+		var err error
+		clusterBootstrapFinished, err = isBootstrapComplete(client)
+		if err != nil {
+			return nil, err
+		}
+
+		webhookDaemonSet := &appsv1.DaemonSet{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "DaemonSet",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+		}
+		nsn := types.NamespacedName{Namespace: NetworkNodeIdentityNamespace, Name: "network-node-identity"}
+		if err := client.Default().CRClient().Get(context.TODO(), nsn, webhookDaemonSet); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to retrieve existing network-node-identity daemonset: %w", err)
+			} else {
+				klog.Infof("network-node-identity daemonset does not exist")
+			}
+		} else {
+			webhookReady = !daemonSetProgressing(webhookDaemonSet, false)
+		}
+
+		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, "network/node-identity/self-hosted"))
+	}
+
+	var webhookCA []byte
+	if err := webhookCAClient.CRClient().Get(context.TODO(), webhookCALookup, webhookCAConfigMap); err != nil {
+		// If the CA doesn't exist, the ValidatingWebhookConfiguration will not be rendered
```

Comment:
why it won't be rendered?

Reply:
I think we don't want to redner the webhook configuration without a CA specified as this is not a valid configuration.
This would mean that if there is an existing config applied it will still be used.